### ISSUE TITLE
Can the dependencies be explained a little bit more?

### ISSUE THREAD

**Issue by nonopolarity** [addyosmani/backbone-fundamentals#199]
It seems that the bare minimum usually is

underscore.js
backbone.js
jquery.js

and backbone.js depends on underscore.js

could this or the overall dependencies be described (unless it is described some where else).

(also about, what if underscore.js isn't used for templating but some other template engine is used, then do we still need to always load underscore.js regardless?)

**Comment by addyosmani** [comment]
Certainly. I'll write something for this soon.

**Comment by addyosmani** [comment]
@jianlin Does this work for you? Just a draft.

https://gist.github.com/4054642

Where would you expect to read this? Given that it covers models, collections etc it might have to go after those sections are explained, although i can alternatively try fitting them in before.

**Comment by addyosmani** [comment]
@g6scheme do you think my draft is sufficient for this? :)

**Comment by gschema** [comment]
:) yup

**Comment by nonopolarity** [comment]
@addyosmani  so is it underscore.js + (jquery or zepto) ?  I sort of expect to see it in the introduction and it can be just two sentences, and then it can point to the code explanation, where there would be more explanation in details.

**Comment by gschema** [comment]
Hey @jianlin,
when in book or documenation is said that Backbone has "hard dependency" on underscore.js and jquery (or zepto) it means that Backbone cannot work without those two libraries as within itself Backbone uses functionality (functions/methods) provided by those libraries. So, if one doesn't include those two libraries Backbone would throw JavaScript errors all around and literary would be unusable (as it would be trying to use undefined functions).

About those libraries:
- Underscore is a utility library for JavaScript that provides a lot of the functional programming support. It is developed by the same guys that developed Backbone. You can read more about it [here](http://underscorejs.org/) and [here](https://github.com/documentcloud/underscore).
- jQuery - A fast, concise, library that simplifies how to traverse HTML documents, handle events, perform animations, and add AJAX. You can read more about it [here](http://jquery.com/) and [here](http://jqfundamentals.com/).

Zepto basically provide same functionality (API) as jQuery, so that makes them swappable.

Underscore also has alternative called [lodash](http://lodash.com/) so it could be swapped with it as well. It is a bit better performing alternative.

Those libraries are swappable as they have the same API, which means that if one library has method (function) called "doSomething()" its alternative has a method with the same name. That way if "doSomething()" is called within Backbone it doesn't care which of the alternatives is used to support it as long as it provides the "doSomething()" method/function.

I hope that this makes sense to you or that I haven't oversimplified.

Best,
Dusan

**Comment by addyosmani** [comment]
@g6scheme Great answer :) Is there anything you would like to add to https://gist.github.com/4054642 before we bring it in? 

I'm trying to figure out where best this should exist - under intro, fundamentals or internals. Internals focuses on components, whilst fundamentals on MVC. I wonder if it should go before the fast facts.

**Comment by nonopolarity** [comment]
@g6scheme and @addyosmani 

The documentation -- do you mean the lines:

> Backbone's only hard dependency is either Underscore.js ( > 1.4.3) or Lo-Dash. For RESTful persistence, history support via Backbone.Router and DOM manipulation with Backbone.View, include json2.js, and either jQuery ( > 1.4.2) or Zepto.

It is a bit hard to decipher at first -- does it mean:

```
// let's say DEPEND_ON is like an "import" or "require" statement:

DEPEND_ON  (Underscore.js ( > 1.4.3)  OR  Lo-Dash);   

if (NEED_TO_USE(RESTful persistence  
                  OR  history support using Backbone.Router  
                  OR  DOM manipulation using Backbone.View)) {
    DEPEND_ON(json2.js);
    DEPEND_ON(jquery.js  OR  Zepto);
}
```

?

**Comment by addyosmani** [comment]
By documentation I was referring to https://gist.github.com/4054642. The line in your comment is actually from the Backbone.js docs, although I do agree it could be more clear. Perhaps we could add a line after we quote this where we say:

"This means that in order to get the most out of Backbone.js, you will also need to include Underscore and jQuery (or equivalent alternatives). Without these dependencies, you will not be able to persist data or create new views."

**Comment by nonopolarity** [comment]
What about the part in the doc that says "include json2.js"... doesn't it need to be explicitly included?

**Comment by addyosmani** [comment]
Resolved in a387f4bd68b1bed618ea27e7056d6ad97c098914
